pandas/tests/io/formats/test_to_csv.py

```diff
@@ -561,3 +563,19 @@ def test_to_csv_compression(self, compression_only,
             result = pd.read_csv(path, index_col=0,
                                  compression=read_compression)
             tm.assert_frame_equal(result, df)
+
+    def test_to_csv_na_rep_long_string(self, capsys):
+        # see gh-25099
+        df = pd.DataFrame({"c": [float('nan')] * 3})
+        df = df.astype("Int64")
+        expected_rows = ['c',
+                         'mynull',
+                         'mynull',
+                         'mynull']
+        expected_ascii = tm.convert_rows_list_to_csv_str(expected_rows)
+
+        df.to_csv(sys.stdout, index=False, na_rep='mynull', encoding='ascii')
+        captured = capsys.readouterr()
```

Comment:
you need to use StringIO here

Reply:
Oh, I think I missed it when squashing commits and merging, even though I left the line `from pandas.compat import StringIO` at the top. Thanks for the spot. Let me fix this one now before I forget it again.